### ISSUE TITLE
Mandatory service configuration in current development branch

### ISSUE THREAD

**Issue by TMiguelT** [broadinstitute/cromwell#4577]
Cromwell: Development (`37-3b2affa`)
Backend: HPC (`ConfigBackendLifecycleActorFactory`)

I wanted access to a recently merged pull request (#4437), so I built a development version of Cromwell. However, when I run it with the same configuration file as I used for Cromwell 36, I get this error:
```
[ERROR] [01/24/2019 11:10:24.126] [cromwell-system-akka.actor.default-dispatcher-4] [akka://cromwell-system/user/SingleWorkflowRunnerActor] No configuration setting found for key 'services'               
akka.actor.ActorInitializationException: akka://cromwell-system/user/SingleWorkflowRunnerActor/ServiceRegistryActor: exception during creation                                                              
        at akka.actor.ActorInitializationException$.apply(Actor.scala:193)                                                                                                                                  
        at akka.actor.ActorCell.create(ActorCell.scala:669)                                                                                                                                                 
        at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:523)                                                                                                                                            
        at akka.actor.ActorCell.systemInvoke(ActorCell.scala:545)                                                                                                                                           
        at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:283)                                                                                                                                
        at akka.dispatch.Mailbox.run(Mailbox.scala:224)                                                                                                                                                     
        at akka.dispatch.Mailbox.exec(Mailbox.scala:235)                                                                                                                                                    
        at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)                                                                                                                                
        at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)                                                                                                                    
        at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)                                                                                                                            
        at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)                                                                                                                   
Caused by: com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'services'                                                                                                   
        at com.typesafe.config.impl.SimpleConfig.findKeyOrNull(SimpleConfig.java:156)                                                                                                                       
        at com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:174)                                                                                                                          
        at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:188)                                                                                                                                
        at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:193)                                                                                                                                
        at com.typesafe.config.impl.SimpleConfig.getObject(SimpleConfig.java:268)                                                                                                                           
        at com.typesafe.config.impl.SimpleConfig.getObject(SimpleConfig.java:41)                                                                                                                            
        at cromwell.services.ServiceRegistryActor$.serviceNameToPropsMap(ServiceRegistryActor.scala:35)                                                                                                     
        at cromwell.services.ServiceRegistryActor.serviceProps(ServiceRegistryActor.scala:63)                                                                                                               
        at cromwell.services.ServiceRegistryActor.<init>(ServiceRegistryActor.scala:65)                                                                                                                     
        at cromwell.services.ServiceRegistryActor$.$anonfun$props$1(ServiceRegistryActor.scala:25)                                                                                                          
        at akka.actor.TypedCreatorFunctionConsumer.produce(IndirectActorProducer.scala:87)                                                                                                                  
        at akka.actor.Props.newActor(Props.scala:212)                                                                                                                                                       
        at akka.actor.ActorCell.newActor(ActorCell.scala:624)                                                                                                                                               
        at akka.actor.ActorCell.create(ActorCell.scala:650)                                                                                                                                                 
        ... 9 more          
```

If I add in a `services` stanza, though, it asks me to define the class of each service, even though they should probably have default values:
```
[ERROR] [01/24/2019 11:09:59.741] [cromwell-system-akka.dispatchers.service-dispatcher-10] [akka://cromwell-system/user/SingleWorkflowRunnerActor/ServiceRegistryActor] Received ServiceRegistryMessage requ
esting service 'LoadController' for which no service is configured.  Message: LoadMetric(NonEmptyList(CallCacheWriteActor),NormalLoad) 
[ERROR] [01/24/2019 11:09:59.731] [cromwell-system-akka.dispatchers.service-dispatcher-10] [akka://cromwell-system/user/SingleWorkflowRunnerActor/ServiceRegistryActor] Received ServiceRegistryMessage requ
esting service 'Instrumentation' for which no service is configured.  Message: InstrumentationServiceMessage(CromwellGauge(CromwellBucket(List(job),NonEmptyList(callcaching, read, $y, queue)),0))
```

***

Here's my config file for Cromwell 36 (that works):
```
backend {
  default = spartan

  providers {
    spartan {
      actor-factory = "cromwell.backend.impl.sfs.config.ConfigBackendLifecycleActorFactory"
      config {
        runtime-attributes = """
        Int runtime_minutes = 600
        Int cpus = 2
        Int requested_memory_mb_per_core = 8000
        String? docker
        """

        submit = """
            sbatch -J ${job_name} -D ${cwd} -o ${out} -e ${err} -t ${runtime_minutes} \
            ${"-c " + cpus} \
            --mem-per-cpu=${requested_memory_mb_per_core} \
            --wrap "/bin/bash ${script}"
        """

        submit-docker = """
            module load Singularity/2.5.0-intel-2017.u2 || true
            singularity pull docker://${docker}
            sbatch -J ${job_name} -D ${cwd} -o ${out} -e ${err} -t ${runtime_minutes} \
            ${"-c " + cpus} \
            --mem-per-cpu=${requested_memory_mb_per_core} \
            --wrap "singularity exec -B ${cwd}:${docker_cwd} docker://${docker} ${job_shell} ${script}"                                                                                                    
        """

        kill = "scancel ${job_id}"
        check-alive = "squeue -j ${job_id}"
        job-id-regex = "Submitted batch job (\\d+).*"
      }
    }
  }
}
```

Here's what I added to my config for 37 that causes the missing class errors:
```
services {                                                                                                                                                                                                  
  MetadataService {                                                                                                                                                                                         
    class = "cromwell.services.metadata.impl.MetadataServiceActor"
  }
}                           
```

**Comment by cjllanwarne** [comment]
Hey @TMiguelT, could you maybe say how you built Cromwell from `develop`?

As part of the build process, `sbt` compiles the `reference.conf` into the jar so that most people don't need to supply anything (if they're happy with the defaults. Is it possible that you compiled in a different way (and so your jar does not contain `reference.conf`)?

**Comment by cjllanwarne** [comment]
(for reference, the way you probably want to be building Cromwell from source is to run `sbt assembly` from the root of the checked-out repository.

**Comment by TMiguelT** [comment]
Yep that's exactly what I did (I thought)

**Comment by kshakir** [comment]
Hm. This looks like a conf bug on our side, but [is your config file importing application.conf](https://cromwell.readthedocs.io/en/stable/tutorials/ConfigurationFiles/#creating-your-first-configuration-file)? That file contains other overrides that cromwell should have over the default `akka` configuration.

The bug here is that application.conf is only supposed to contain overrides, while reference.conf should contain newly defined resources. Since the `services` block are cromwell's services, they should be newly defined in reference.conf. That would then allow anyone who accidentally doesn't pick up our application.conf to *at least* have the reference `services`, plus the original `akka` values with degraded cromwell performance.

**Comment by kshakir** [comment]
Fixed in #4589
